modules/distr-tracing-otel-migrating-from-jaeger-with-sidecars.adoc

```diff
@@ -0,0 +1,110 @@
+
+The {OTELShortName} Operator supports sidecar injection into Deployment workloads, so you can migrate from a {JaegerShortName} sidecar to a {OTELShortName} sidecar.
+
+.Prerequisites
+
+* The {JaegerName} is used on the cluster.
+* The {OTELName} is installed.
+
+.Procedure
+
+. Configure the OpenTelemetry Collector as a sidecar.
++
+[source,yaml]
+----
+apiVersion: opentelemetry.io/v1alpha1
+kind: OpenTelemetryCollector
+metadata:
+  name: otel
+  namespace: <otel-collector-namespace>
+spec:
+  mode: sidecar
+  config: |
+    receivers:
+      jaeger:
+        protocols:
+          grpc:
+          thrift_binary:
+          thrift_compact:
+          thrift_http:
+    processors:
+      batch:
+      memory_limiter:
+        check_interval: 1s
+        limit_percentage: 50
+        spike_limit_percentage: 30
+      resourcedetection:
+        detectors: [openshift]
+        timeout: 2s
+    exporters:
+      otlp:
+        endpoint: "tempo-<example>-gateway:8090" <1>
+        tls:
+          insecure: true
+    service:
+      pipelines:
+        traces:
+          receivers: [jaeger]
+          processors: [memory_limiter, resourcedetection, batch]
+          exporters: [otlp]
+----
+<1> This endpoint points to the Gateway of a TempoStack instance deployed by using the `<example>` {TempoOperator}.
+
+. Create a service account for running your application.
++
+[source,yaml]
+----
+apiVersion: v1
+kind: ServiceAccount
+metadata:
+  name: otel-collector-sidecar
+----
+
+. Create a cluster role for the permissions needed by some processors.
++
+[source,yaml]
+----
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRole
+metadata:
+  name: otel-collector-sidecar
+rules:
+  <1>
+- apiGroups: ["config.openshift.io"]
+  resources: ["infrastructures", "infrastructures/status"]
+  verbs: ["get", "watch", "list"]
+----
+<1> The `resourcedetectionprocessor` requires permissions for infrastructures and infrastructures/status.
+
+. Create a `ClusterRoleBinding` to set the permissions for the service account.
++
+[source,yaml]
+----
+apiVersion: rbac.authorization.k8s.io/v1
+kind: ClusterRoleBinding
+metadata:
+  name: otel-collector-sidecar
+subjects:
+- kind: ServiceAccount
+  name: otel-collector-deployment
+  namespace: otel-collector-example
+roleRef:
+  kind: ClusterRole
+  name: otel-collector
+  apiGroup: rbac.authorization.k8s.io
+----
+
+. Deploy the OpenTelemetry Collector as a sidecar.
+
+. Remove the injected Jaeger Agent from your application by removing the `"sidecar.jaegertracing.io/inject": "true"` annotation from your Deployment.
+
+. Enable automatic injection of the OpenTelemetry sidecar by adding the `sidecar.opentelemetry.io/inject: "true"` annotation to the `.spec.template.metadata.annotations` field of your Deployment.
+
+. Use the created service account for the Deployment of your application to allow the processors to get the correct information and add it to your traces.
```

Comment:
"deployment" or "`Deployment` object"
```suggestion
. Use the created service account for the deployment of your application to allow the processors to get the correct information and add it to your traces.
```

Reply:
Thank you. Applied manually. I also corrected this in all the other files.